python/paddle/fluid/trainer.py

```diff
@@ -68,22 +68,32 @@ def __init__(self, program_func, optimizer, param_path=None, place=None):
         # 1. we need to generate a framework.Program by calling
         # program_func. Reference: fluid.program_guard in
         # test_word2vec.py
+        self.predict_vars = None
         self.scope = core.Scope()
 
         self.startup_program = framework.Program()
         self.train_program = framework.Program()
 
         with framework.program_guard(self.train_program, self.startup_program):
-            loss = program_func()
             if not isinstance(optimizer, opt_module.Optimizer):
                 raise TypeError(
                     "The optimizer should be an instance of Optimizer")
+            out = program_func()
+            if not isinstance(out, tuple) or len(out) != 2:
+                raise ValueError(
+                    "program_func should return tuple(loss, predict_vars)")
+
+            loss = out[0]
```

Comment:
Here is a problem I also encounter in my `trainer.test()`. The `program_func()` may returns more than one element. But only one of them('loss') will be used to build the backward part. In the current design, we have no method to point out which one is the loss.

Reply:
I create a issue for this problem https://github.com/PaddlePaddle/Paddle/issues/10474.